luminoth/eval.py

```diff
@@ -4,35 +4,41 @@
 import tensorflow as tf
 import time
 
-from luminoth.datasets import TFRecordDataset
-from luminoth.models import get_model
+from luminoth.datasets import get_dataset
+from luminoth.models import (
+    get_model, DEFAULT_MODEL
+)
 from luminoth.utils.config import (
-    get_model_config
+    get_model_config, load_config
 )
 from luminoth.utils.bbox_overlap import bbox_overlap
 from luminoth.utils.image_vis import image_vis_summaries
 
 
 @click.command(help='Evaluate trained (or training) models')
-@click.option('model_type', '--model', required=True, default='fasterrcnn')
 @click.option('dataset_split', '--split', default='val', help='Dataset split to use.')  # noqa
-@click.option('config_file', '--config', '-c', help='Config to use.')  # noqa
+@click.option('config_files', '--config', '-c', required=True, multiple=True, help='Config to use.')  # noqa
 @click.option('--job-dir', required=True, help='Directory from where to read saved models and write evaluation logs.')  # noqa
 @click.option('--watch/--no-watch', default=True, help='Keep watching checkpoint directory for new files.')  # noqa
 @click.option('--from-global-step', type=int, default=None, help='Consider only checkpoints after this global step')  # noqa
 @click.option('override_params', '--override', '-o', multiple=True, help='Override model config params.')  # noqa
 @click.option('--image-vis', is_flag=True, default=False, help='Display images in TensorBoard.')  # noqa
 @click.option('--files-per-class', type=int, default=10, help='How many files per class display in every epoch.')  # noqa
-def evaluate(model_type, dataset_split, config_file, job_dir, watch,
+def evaluate(dataset_split, config_files, job_dir, watch,
              from_global_step, override_params, image_vis, files_per_class):
     """
     Evaluate models using dataset.
     """
-    model_cls = get_model(model_type)
-    config = model_cls.base_config
+    custom_config = load_config(config_files)
+    # If the config file is empty, our config will be the base_config for the
+    # default model.
```

Comment:
Does it make sense to have a default model with its own config? Shouldn't the default model be present on the base config file? If not, we should make it mandatory for the user to specify the model in its config file, since as of now `DEFAULT_MODEL` acts as a base config entry that isn't present on a config file, right?

Reply:
Making `model.type` mandatory in the user config (and removing DEFAULT_MODEL) makes sense to me.

> Shouldn't the default model be present on the base config file?

As of now there is no model-independent base config, and I don't know that there should be one. Most options are model-specific (even those outside the `model` key, e.g. `train.learning_rate`, `dataset.data_augmentation` are things that may need to be changed from model to model), more so if we're planning on including non-object-detection models later on.

> since as of now DEFAULT_MODEL acts as a base config entry that isn't present on a config file, right?

Yes. Currently, if `model.type` is not defined in the user config, it gets set to the value in `DEFAULT_MODEL`.